<launch>
  
  <!-- By default we do not overwrite the URDF. Change the following to true to change the default behavior -->
  <arg name="load_robot_description" default="false"/>

  <!-- The name of the parameter under which the URDF is loaded -->
  <arg name="robot_description" default="robot_description"/>


  <include file="$(find sr_moveit_hand_config)/launch/test_load_prefix.launch" />
  
  <!-- The semantic description that corresponds to the URDF -->
  <node name="generate_srdf" pkg="sr_moveit_hand_config" type="generate_srdf.py" respawn="false" output="screen" args="$(find sr_moveit_hand_config)/config/shadowhands.xacro.srdf" />
    
  <!--  blocks the other param loader and deadlocks on robotdescription missing 
  <param name="$(arg robot_description)_semantic" command="$(find sr_moveit_hand_config)/scripts/generate_srdf.py  $(find sr_moveit_hand_config)/config/shadowhand_motor.srdf" />
  -->

  <!-- Load updated joint limits (override information from URDF) -->
  <group ns="$(arg robot_description)_planning">
    <node name="load_joint_limits" pkg="sr_moveit_hand_config" type="generate_load_moveit_config.py" respawn="false" output="screen" args="joint_limits $(find sr_moveit_hand_config)/config/joint_limits_template.yaml" />
  </group>

  <!-- Load default settings for kinematics; these settings are overridden by settings in a node's namespace -->
  <group ns="$(arg robot_description)_kinematics">
    <node name="load_kinematics" pkg="sr_moveit_hand_config" type="generate_load_moveit_config.py" respawn="false" output="screen" args="kinematics $(find sr_moveit_hand_config)/config/kinematics_template.yaml"/>
  </group>
  <!-- Load fake controllers params-->
  <group ns="$(arg robot_description)_planning">
    <node name="load_fake_controllers" pkg="sr_moveit_hand_config" type="generate_load_moveit_config.py" respawn="false" output="screen" args="fake_controllers"/>
  </group>
  <!-- ompl planning params -->
  <node name="load_ompl_planning" pkg="sr_moveit_hand_config" type="generate_load_moveit_config.py" respawn="false" output="screen" args="ompl_planning $(find sr_moveit_hand_config)/config/ompl_planning_template.yaml"/>



</launch>
